### PR TITLE
feat: direct allocation strategy & direct allocated event handler

### PR DESCRIPTION
**Pull request by 0xnigir1**
# 🤖 Linear

Closes GIT-159

## Description
- `DirectAllocationStrategy` handler
- `DirectAllocated` event handler for DirectAllocationStrategy

## Checklist before requesting a review

-   [x] I have conducted a self-review of my code.
-   [x] I have conducted a QA.
-   [x] If it is a core feature, I have included comprehensive tests.

### CODE REVIEW

File: packages/processors/src/processors/strategy/directAllocation/directAllocation.handler.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    async handle(event: ProcessorEvent<"Strategy", StrategyEvent>): Promise<Changeset[]> {
+        switch (event.eventName) {
+            case "DirectAllocated":
+                return new DirectAllocatedHandler(
+                    event as ProcessorEvent<"Strategy", "DirectAllocated">,
+                    this.chainId,
+                    this.dependencies,
+                ).handle();
+            default:
+                throw new UnsupportedEventException("Strategy", event.eventName);
```
Comment (0xkenj1) [inline packages/processors/src/processors/strategy/directAllocation/directAllocation.handler.ts]: we improve this error by providing strategyName as optional or maybe create a new custom error for strategies. wdyt ?

File: packages/processors/src/processors/strategy/directAllocation/handlers/directAllocated.handler.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        const strategyAddress = getAddress(this.event.srcAddress);
+
+        const round = await roundRepository.getRoundByStrategyAddressOrThrow(
+            this.chainId,
+            strategyAddress,
+        );
+        const project = await projectRepository.getProjectByIdOrThrow(
+            this.chainId,
+            this.event.params.profileId,
+        );
```
Comment (0xkenj1) [inline packages/processors/src/processors/strategy/directAllocation/handlers/directAllocated.handler.ts:52]: sweet

File: packages/processors/src/processors/strategy/directAllocation/directAllocation.handler.ts (hunk 1 above)
Comment (0xnigir1) [reply packages/processors/src/processors/strategy/directAllocation/directAllocation.handler.ts]: i buy the first option 🫡

**Review by 0xkenj1 — APPROVED** [review]
God

File: packages/processors/src/processors/strategy/directAllocation/directAllocation.handler.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { Changeset } from "@grants-stack-indexer/repository";
+import { ChainId, ProcessorEvent, StrategyEvent } from "@grants-stack-indexer/shared";
+
+import { ProcessorDependencies, UnsupportedEventException } from "../../../internal.js";
+import { BaseStrategyHandler } from "../index.js";
+import { DirectAllocatedHandler } from "./handlers/index.js";
+
+const STRATEGY_NAME = "allov2.DirectAllocationStrategy";
```
Comment (jahabeebs) [inline packages/processors/src/processors/strategy/directAllocation/directAllocation.handler.ts:8]: as we add more of these we should extract these into one place

**Review by jahabeebs — APPROVED** [review]
🎉

File: packages/processors/src/processors/strategy/directAllocation/directAllocation.handler.ts (hunk 3 above)
Comment (0xnigir1) [reply packages/processors/src/processors/strategy/directAllocation/directAllocation.handler.ts:8]: yeah probably, one more to go (for now)

**Pull request merged**